### PR TITLE
cellGame /cellMsgDialog stuff

### PR DESCRIPTION
**Pull request by Megamouse**
- moved cell message dialogs to their own function
- simplified cellGameContentErrorDialog
- implemented cellHddGameExitBroken and cellGameDataExitBroken (https://github.com/RPCS3/rpcs3/issues/4743)
- added more param checks: [Results cellGame.txt](https://github.com/RPCS3/rpcs3/files/2095695/Results.cellGame.txt)

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellGame.h [hunk 1]
```diff
@@ -57,19 +57,19 @@ enum
...
-	CELL_GAME_GAMETYPE_SYS  = 0,
-	CELL_GAME_GAMETYPE_DISC = 1,
-	CELL_GAME_GAMETYPE_HDD  = 2,
-	CELL_GAME_GAMETYPE_HOME = 4,
-
+	CELL_GAME_GAMETYPE_SYS      = 0,
+	CELL_GAME_GAMETYPE_DISC     = 1,
 	CELL_GAME_GAMETYPE_GAMEDATA = 3,
+	CELL_GAME_GAMETYPE_HDD      = 2,
+	CELL_GAME_GAMETYPE_HOME     = 4,
```
Comment (kd-11) [inline rpcs3/Emu/Cell/Modules/cellGame.h]: if you're going to reorder enums, why go with 0,1,3,2,4? Triggered

File: rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp [hunk 2]
```diff
@@ -15,10 +15,8 @@ MsgDialogBase::~MsgDialogBase()
 {
 }
 
-s32 cellMsgDialogOpen2(u32 type, vm::cptr<char> msgString, vm::ptr<CellMsgDialogCallback> callback, vm::ptr<void> userData, vm::ptr<void> extParam)
+s32 open_msg_dialog(u32 type, vm::cptr<char> msgString, vm::ptr<CellMsgDialogCallback> callback, vm::ptr<void> userData, vm::ptr<void> extParam)
```
Comment (kd-11) [inline rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: By making this an emulator internal method, it no longer feels like it belongs in cellMsgDialog (same way trophynotificationhelper exists outside the trophies implementation). It has now become an internal utility function. Not a blocker for merge but it feels wierd now.

File: rpcs3/Emu/Cell/Modules/cellGame.h (hunk 1 above)
Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellGame.h]: Ugh good catch

File: rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp (hunk 2 above)
Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: It was the option with the least loc

Comment (kd-11) [reply rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: ¯\_(ツ)_/¯

**Pull request merged**